libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -784,6 +797,9 @@ method publish*(g: GossipSub, topic: string, data: seq[byte]): Future[int] {.asy
 
   g.broadcast(peers, RPCMsg(messages: @[msg]), isHighPriority = true)
 
+  if isLargeMessage(msg, msgId):
+    g.sendIDontWant(msg, msgId, peers)
```

Comment:
... why should we send IDONTWANT _after_ broadcasting the message? If the other peers receive the message, they already know that I don't want it.

There exists a case where sending it before broadcasting makes sense, though I'm not sure how well it has been studied, ie it needs research: when streaming a large message to a peer, this takes time - in the meantime, the peer might start streaming the same message to _us_ - an IDONTWANT before sending the message tells the other peer not to do that.